### PR TITLE
[19236] Fix bitmask code generation

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This PR supersedes #205

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
[Manual CI action](https://github.com/eProsima/Fast-DDS-Gen/actions/runs/6160288973)

File: src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg [hunk 1]
```diff
@@ -148,7 +148,33 @@ int $if(union.hasScope)$$union.scope$::$endif$compare$union.name$(
 
 enum_type(ctx, parent, enum) ::= <<>>
 
-bitmask_type(ctx, parent, bitmask) ::= <<>>
+bitmask_type(ctx, parent, bitmask) ::= <<
+void $if(bitmask.hasScope)$$bitmask.scope$::$endif$print$bitmask.name$(
+        $bitmask.name$* topic)
+{
+    printf("$bitmask.name$: { \n");
+    $bitmask.members:{ it | printf("$it.name$: %s\n", *topic & $bitmask.name$Bits::$it.name$ == $bitmask.name$Bits::$it.name$ ? "true" : "false");}; separator="\n"$
```
Comment (richiware) [inline src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg:156]:
```suggestion
    $bitmask.members:{ member | printf("$member.name$: %s\n", *topic & $bitmask.name$Bits::$member.name$ == $bitmask.name$Bits::$member.name$ ? "true" : "false");}; separator="\n"$
```

File: src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg [hunk 2]
```diff
@@ -148,7 +148,33 @@ int $if(union.hasScope)$$union.scope$::$endif$compare$union.name$(
...
+}
+
+void $if(bitmask.hasScope)$$bitmask.scope$::$endif$initialize$bitmask.name$(
+        $bitmask.name$* topic)
+{
+    $bitmask.members:{ it |
```
Comment (richiware) [inline src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg:162]:
```suggestion
    $bitmask.members:{ member |
```

File: src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg [hunk 3]
```diff
@@ -148,7 +148,33 @@ int $if(union.hasScope)$$union.scope$::$endif$compare$union.name$(
...
+    if (rand() % 2 == 1)
+    {
+        *topic |= $bitmask.name$Bits::$it.name$;
```
Comment (richiware) [inline src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg:165]:
```suggestion
        *topic |= $bitmask.name$Bits::$member.name$;
```

**Review by richiware — CHANGES_REQUESTED** [review]

**Review by richiware — CHANGES_REQUESTED** [review]
bitmask members should be initialized to 0. Right now, when a user creates an structure with a bitmask member, this one contains garbage.

File: src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg:156]: TOC :smirk:

File: src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg (hunk 2 above)
Comment (JLBuenoLopez-eProsima) [reply src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg:162]: :smirk:

**Comment by JLBuenoLopez-eProsima** [comment]
> bitmask members should be initialized to 0. Right now, when a user creates an structure with a bitmask member, this one contains garbage.

You have just asked for an IDL-Parser v1.7.2 release! :rofl: The initial value was removed in:

* eProsima/IDL-Parser#79

**Comment by JLBuenoLopez-eProsima** [comment]
[Manual CI action](https://github.com/eProsima/Fast-DDS-Gen/actions/runs/6169028818)

File: src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg [hunk 4]
```diff
@@ -148,7 +148,33 @@ int $if(union.hasScope)$$union.scope$::$endif$compare$union.name$(
 
 enum_type(ctx, parent, enum) ::= <<>>
 
-bitmask_type(ctx, parent, bitmask) ::= <<>>
+bitmask_type(ctx, parent, bitmask) ::= <<
+void $if(bitmask.hasScope)$$bitmask.scope$::$endif$print$bitmask.name$(
+        $bitmask.name$* topic)
+{
+    printf("$bitmask.scopedname$: { \n");
+    $bitmask.members:{ member | printf("$member.name$: %s\n", *topic & $bitmask.name$Bits::$member.name$ == $bitmask.name$Bits::$member.name$ ? "true" : "false");}; separator="\n"$
```
Comment (richiware) [inline src/main/java/com/eprosima/fastdds/idl/templates/SerializationSource.stg:156]: Print always all values true or all values false. Fix:

```suggestion
    $bitmask.members:{ member | printf("$member.name$: %s\n", ((*topic & $bitmask.name$Bits::$member.name$) == $bitmask.name$Bits::$member.name$) ? "true" : "false");}; separator="\n"$
```

**Review by richiware — CHANGES_REQUESTED** [review]

**Review by richiware — APPROVED** [review]

**Pull request merged**